### ISSUE TITLE
kustomize edit set image removes replicas

### ISSUE THREAD

**Issue by jskold** [kubernetes-sigs/kustomize#1486]
In Kustomize 3.1.0, running `kustomize edit set image` removes the replicas block from kustomization.yaml.

Sample to reproduce this bug:

*kustomize.yaml*
```yaml
apiVersion: kustomize.config.k8s.io/v1beta1
kind: Kustomization
namespace: test
resources:
- deployment.yaml
replicas:
- name: nginx-deployment
  count: 2
```
*deployment.yaml*
```yaml
kind: Deployment
apiVersion: apps/v1
metadata:
  name: nginx-deployment
spec:
  template:
    spec:
      containers:
        - name: nginx
          image: nginx
```

Running `kustomize edit set image nginx-deployment:latest` rewrites the kustomization.yaml so that it looks like this (note the missing replicas block):

*kustomization.yaml*
```yaml
apiVersion: kustomize.config.k8s.io/v1beta1
kind: Kustomization
namespace: test
resources:
- deployment.yaml
images:
- name: nginx
  newTag: latest
```

**Comment by richardmarshall** [comment]
Hi @jskold, this bug was fixed in #1432 and will be included in the next release.

**Comment by jskold** [comment]
Thanks @richardmarshall!
